### PR TITLE
feat: Add 'Disable on Prefill' option for fields

### LINKED ISSUE #15712: [CAL-4007] booking questions: ability to disabled input if url is prefilled
give people the option to lock certain identifiers and if the URL parameter exists, disable the input:

image

if this is active, and the link is: cal.com/peer/meet?**email=john@example.com** it should lock the email input.

CleanShot 2024-07-09 at 20 43 32@2x

ideally we can lock every identifier (also custom questions)

<sub>[CAL-4007](https://linear.app/calcom/issue/CAL-4007/booking-questions-ability-to-disabled-input-if-url-is-prefilled)</sub>

### PR DESCRIPTION
**Pull request by rajesh-jonnalagadda**
/claim #15712 

## What does this PR do?

Added a new checkbox to disable the input. this change provides the user the option to lock certain identifiers and if the URL parameter exists, disable the input: 

- Fixes #15712  (GitHub issue number)
- Fixes CAL-4007 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://www.loom.com/share/b09fb51e46194115b89c81f0431cb500

Demo Loom with all fields https://www.loom.com/share/63f34b4506564efeb124abcadb520489

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
N/A
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
1. create a new event or take any event of your choice.
2. click on the event
3. go to advanced
4. In question section. add a new question.(identifier: testprefill)
5. The question form should have a check box with the description (Disable input if the URL identifier is prefilled)
6. check the checkbox as mentioned earlier.
7. fill in other details and save.
8. update the event by clicking the save button at the top right corner.
9. Do it for any existing question(for example email)
10. Now click on Preview at the top right side.
11. fill in the month and time and submit the form
12. now edit the URL by adding new search params(&testprefill=test&email=test@gmail.com) and enter
13. The form should be replaced with the URL search parameters.
14. Email and test refill input should be disabled.
15. click on the submit button.

- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/public/static/locales/en/common.json [hunk 1]
```diff
@@ -2498,5 +2498,6 @@
   "USER_MEMBER_OF_OTHER_ORGANIZATION": "User is member of an organization that this team is not a part of.",
   "rescheduling_not_possible": "Rescheduling is not possible as the event has expired",
   "event_expired": "This event is expired",
-  "ADD_NEW_STRINGS_ABOVE_THIS_LINE_TO_PREVENT_MERGE_CONFLICTS": "↑↑↑↑↑↑↑↑↑↑↑↑↑ Add your new strings above here ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑"
+  "ADD_NEW_STRINGS_ABOVE_THIS_LINE_TO_PREVENT_MERGE_CONFLICTS": "↑↑↑↑↑↑↑↑↑↑↑↑↑ Add your new strings above here ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑",
+  "pre_fill_check": "Disable input if the URL identifier is prefilled"
```
Comment (anikdhabal) [inline apps/web/public/static/locales/en/common.json:2502]: Add the i18nkey above `ADD_NEW_STRINGS_ABOVE_THIS_LINE` key

**Review by anikdhabal — CHANGES_REQUESTED** [review]

File: apps/web/public/static/locales/en/common.json (hunk 1 above)
Comment (rajeshj11) [reply apps/web/public/static/locales/en/common.json:2502]: ack

**Comment by rajeshj11** [comment]
@anikdhabal could  you please take a look at the updated pr.

File: apps/api/v2/src/ee/event-types/event-types_2024_04_15/inputs/update-event-type.input.ts [hunk 2]
```diff
@@ -177,6 +177,10 @@ export class BookingField_2024_04_15 {
   @ValidateNested({ each: true })
   @Type(() => Source)
   sources?: Source[];
+
+  @IsOptional()
+  @IsBoolean()
+  prefillChecked?: boolean;
```
Comment (hariombalhara) [inline apps/api/v2/src/ee/event-types/event-types_2024_04_15/inputs/update-event-type.input.ts:183]: Why don't we call it `disableOnPrefill`?

File: apps/web/public/static/locales/en/common.json [hunk 3]
```diff
@@ -2496,5 +2496,6 @@
   "USER_MEMBER_OF_OTHER_ORGANIZATION": "User is member of an organization that this team is not a part of.",
   "rescheduling_not_possible": "Rescheduling is not possible as the event has expired",
   "event_expired": "This event is expired",
+  "pre_fill_check": "Disable input if the URL identifier is prefilled",
```
Comment (hariombalhara) [inline apps/web/public/static/locales/en/common.json:2499]: Could be called, `disable_input_if_prefilled`

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx [hunk 4]
```diff
@@ -133,6 +135,10 @@ export const BookingFields = ({
           });
         }
 
+        if (field.prefillChecked && searchParams && searchParams[field.name]) {
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:138]: FormBuilder logic must be contained within FormBuilder. This place is suitable only for logic that is specific to Booking Form fields.

File: apps/web/playwright/manage-booking-questions.e2e.ts [hunk 5]
```diff
@@ -262,6 +262,91 @@ test.describe("Manage Booking Questions", () => {
...
 });
 
+test("Disable Input on url prefill for booking and verify a few thing in b/w", async ({
+  page,
+  users,
+  context,
+}, testInfo) => {
+  // Considering there are many steps in it, it would need more than default test timeout
+  test.setTimeout(testInfo.timeout * 2);
+  const user = await createAndLoginUserWithEventTypes({ users, page });
```
Comment (hariombalhara) [inline apps/web/playwright/manage-booking-questions.e2e.ts:272]: I don't think we need `e2e` for this change. e2es are costly time wise and easily become flaky.

So, we are intentionally avoiding e2e wherever possible and focus on Unit tests.


This change is pretty straightforward. I would much prefer to have Unit tests for `FormBuilderField` and `FormBuilder` components instead.

Can be a followup PR as well.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
@rajeshj11 Thank you so much for the PR 🙏 I have left a few suggestions

**Comment by rajeshj11** [comment]
> @rajeshj11 Thank you so much for the PR 🙏 I have left a few suggestions

Thanks for the feedback. Will work on it and update the pr.

File: apps/web/playwright/manage-booking-questions.e2e.ts (hunk 5 above)
Comment (rajeshj11) [reply apps/web/playwright/manage-booking-questions.e2e.ts:272]: @hariombalhara  can you suggest any reference to work with this?

**Review by anikdhabal — DISMISSED** [review]
Hey @rajeshj11, it would be better if you revert the e2e test and unit test, as @hariombalhara suggested. If you encounter any problems, please create a follow-up PR for the test later. Apart from this, please fix the conflicts and address the requested changes made by Hariom. Thank you.

**Comment by rajeshj11** [comment]
> Hey @rajeshj11, it would be better if you revert the e2e test and unit test, as @hariombalhara suggested. If you encounter any problems, please create a follow-up PR for the test later. Apart from this, please fix the conflicts and address the requested changes made by Hariom. Thank you.

updated the pr with feedback changes. will handle unit test cases later. Please review
demo video
https://www.loom.com/share/4f6dffaf2661455e9fd4ee7640d577fb

File: apps/web/playwright/manage-booking-questions.e2e.ts (hunk 5 above)
Comment (hariombalhara) [reply apps/web/playwright/manage-booking-questions.e2e.ts]: You an look at some of *.test.tsx files that have react-testing-library based tests for components
image

File: apps/web/playwright/manage-booking-questions.e2e.ts [hunk 6]
```diff
@@ -546,6 +546,10 @@ async function addQuestionAndSave({
     await page.fill('[name="name"]', question.name);
   }
 
+  if (question.disableOnPrefill !== undefined) {
+    await page.click('[name="disableOnPrefill"]');
+  }
```
Comment (hariombalhara) [inline apps/web/playwright/manage-booking-questions.e2e.ts:551]: Why this change?
This feature doesn't change the default behaviour so e2e tests shouldn't have a need to be modified.

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx [hunk 7]
```diff
@@ -29,7 +29,6 @@ export const BookingFields = ({
   const locationResponse = watch("responses.location");
   const currentView = rescheduleUid ? "reschedule" : "";
   const isInstantMeeting = useBookerStore((state) => state.isInstantMeeting);
-
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:32]: Unnecessary whitespace.

File: apps/web/playwright/manage-booking-questions.e2e.ts (hunk 5 above)
Comment (rajeshj11) [reply apps/web/playwright/manage-booking-questions.e2e.ts]: > You an look at some of *.test.tsx files that have react-testing-library based tests for components image

thanks a lot. will check and update the pr

File: packages/features/form-builder/FormBuilderField.tsx [hunk 8]
```diff
@@ -53,9 +54,20 @@ export const FormBuilderField = ({
 }) => {
   const { t } = useLocale();
   const { control, formState } = useFormContext();
+  const searchParams = useRouterQuery();
 
   const { hidden, placeholder, label } = getAndUpdateNormalizedValues(field, t);
 
+  const isPreFilledInUrl = (): boolean => {
+    const message = (formState?.errors?.responses?.message || "") as string;
+    const name = message.replace(/\{([^}]+)\}.*/, "$1");
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:63]: Let's abstract it out in a function
```suggestion
    const name = getFieldNameFromErrorMessage(errorMessage)
```

Also, use the function in other place as well from which this regex is copied.

File: packages/features/form-builder/FormBuilderField.tsx [hunk 9]
```diff
@@ -53,9 +54,20 @@ export const FormBuilderField = ({
 }) => {
   const { t } = useLocale();
   const { control, formState } = useFormContext();
+  const searchParams = useRouterQuery();
 
   const { hidden, placeholder, label } = getAndUpdateNormalizedValues(field, t);
 
+  const isPreFilledInUrl = (): boolean => {
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:61]:
```suggestion
  const shouldBeDisabledDueToPrefill = (): boolean => {
```

This seems like a better name as it isn't just checking if the field is prefilled, it is also checking if the field is marked for disabling on prefill.

File: packages/features/form-builder/FormBuilderField.tsx [hunk 10]
```diff
@@ -53,9 +54,20 @@ export const FormBuilderField = ({
...
+    const message = (formState?.errors?.responses?.message || "") as string;
+    const name = message.replace(/\{([^}]+)\}.*/, "$1");
+    //if the field is already prefilled in url and it has error then we don't want to disable it
+    if (name === field.name) {
+      return false;
+    }
+    return !!(field.disableOnPrefill && searchParams && searchParams[field.name]);
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:68]: Could be simplified by doing early return at the top with the following code

```
if (field.disableOnPrefill || !searchParams) {return false}
```
After that it can be simply 
```suggestion
    return !!searchParams[field.name]);
```

File: apps/web/playwright/manage-booking-questions.e2e.ts [hunk 11]
```diff
@@ -530,7 +530,6 @@ async function addQuestionAndSave({
   };
 }) {
   await page.click('[data-testid="add-field"]');
-
```
Comment (hariombalhara) [inline apps/web/playwright/manage-booking-questions.e2e.ts:533]: These are also unnecessary.

File: packages/features/form-builder/FormBuilder.tsx [hunk 12]
```diff
@@ -654,6 +659,11 @@ function VariantFields({
         label={t("identifier")}
       />
 
+      <CheckboxField
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilder.tsx:662]: Added support for name field as well which is a variantsSupportedField and has a bit different UI.

File: packages/features/form-builder/schema.ts [hunk 13]
```diff
@@ -308,12 +309,12 @@ export const fieldTypesSchemaMap: Partial<
         }
         const valueIdentified = response as unknown as Record<string, string>;
         if (subField.required) {
+          if (!isPartialSchema && !valueIdentified[subField.name])
```
Comment (hariombalhara) [inline packages/features/form-builder/schema.ts:312]: Fixed by @hariombalhara.

Invalid String error should be the fallback. We don't want such error if the subField is required and not filled.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Great progress @rajeshj11 ! Left some more comments and fixed a few things that I found while testing

**Comment by rajeshj11** [comment]
@anikdhabal @hariombalhara I have updated the pr with feedback changes. and also added unit tests(needed feedback on those unit tests)

File: packages/features/form-builder/FormBuilderField.tsx [hunk 14]
```diff
@@ -55,6 +59,7 @@ export const FormBuilderField = ({
   const { control, formState } = useFormContext();
 
   const { hidden, placeholder, label } = getAndUpdateNormalizedValues(field, t);
+  const shoudlBeDisabled = useShouldBeDisabledDueToPrefill(field);
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:62]:
```suggestion
  const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
```

File: packages/features/form-builder/FormBuilderField.tsx [hunk 15]
```diff
@@ -68,7 +73,7 @@ export const FormBuilderField = ({
               <ComponentForField
                 field={{ ...field, label, placeholder, hidden }}
                 value={value}
-                readOnly={readOnly}
+                readOnly={readOnly || shoudlBeDisabled}
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:76]:
```suggestion
                readOnly={readOnly || shouldBeDisabled}
```

File: packages/ui/components/formbuilder/fromBuilder.test.tsx
Comment (hariombalhara) [inline packages/ui/components/formbuilder/fromBuilder.test.tsx:1]: Typo in file name. Should FormBuilder.test.tsx.

File: packages/ui/components/formbuilder/utils.ts
Comment (hariombalhara) [inline packages/ui/components/formbuilder/utils.ts:1]: It would be better to call it `testUtils.ts` to convey that it is specific to test.

File: packages/ui/components/formbuilder/testUtils.ts
Comment (hariombalhara) [inline packages/ui/components/formbuilder/testUtils.ts:1]: We keep the tests in the same folder as source code. 

So, testUtils and tests should be in @calcom/features/form-builder/

File: packages/ui/components/formbuilder/formBuilder.test.tsx
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilder.test.tsx:1]: It tests FormBuilder.ts so the name should be FormBuilder.test.tsx with F capital

File: packages/ui/components/formbuilder/formBuilderField.test.tsx
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:1]: The name should be FormBuilderField.test.tsx

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+  { questionType: "textarea", label: "Long Text Field" },
+  { questionType: "select", label: "Select Field" },
+  { questionType: "multiselect", label: "MultiSelect Field" },
+  { questionType: "multiemail", label: "Multiple Emails Field" },
+  { questionType: "checkbox", label: "CheckBox Group Field" },
+  { questionType: "radio", label: "Radio Group Field" },
+  { questionType: "boolean", label: "Checkbox Field" },
+];
+
+const getFormatStateForQuestionType = (questionType: string) => {
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:43]: Why does it say 'Format', I think it should be FormState

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+      const mockFormContext = { formState, control: {} as Control };
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+
+      expect(shouldBeDisabled).toBe(false);
+    });
+  }
+  for (const { questionType, label } of questionTypes) {
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:92]: I don't think it matters to `useShouldBeDisabledDueToPrefill` as to which field it is. It just needs `name` and `disableOnPrefill` from the field, so we can do away with this questions iteration. It is important to keep the tests as simple as possible

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+});
+
+describe("FormBuilderField: Disable on Prefill Behavior", () => {
+  beforeEach(() => {
+    // Reset mocks before each test
+    vi.resetAllMocks();
+  });
+
+  for (const { questionType, label } of questionTypes) {
+    test(`should remain enabled when 'disableOnPrefill' is true and there are errors for ${label} (${questionType})`, () => {
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:72]:
```suggestion
    test(`should return false when 'disableOnPrefill' is `true` but there are errors for ${label} (${questionType})`, () => {
```

File: packages/features/form-builder/FormBuilderField.tsx [hunk 19]
```diff
@@ -42,6 +43,32 @@ type ValueProps =
...
+  const name = errorMessage?.replace(/\{([^}]+)\}.*/, "$1");
+  return name;
+};
+
+export const useShouldBeDisabledDueToPrefill = (field: RhfFormField): boolean => {
+  const { formState } = useFormContext();
+  const searchParams = useRouterQuery();
+  // Get the value of a specific field
+  const errorMessage = (formState?.errors?.responses?.message || "") as string;
+  const name = getFieldNameFromErrorMessage(errorMessage);
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:56]:
```suggestion
  const fieldNameThatHasError = getFieldNameFromErrorMessage(errorMessage);
```

File: packages/features/form-builder/FormBuilderField.tsx [hunk 20]
```diff
@@ -42,6 +43,32 @@ type ValueProps =
...
+  // If a field is prefilled via the URL and an error occurs upon form submission, we should not disable the field.
+  if (!field) {
+    return false;
+  }
+  if (name === field?.name) {
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:61]:
```suggestion
  if (fieldNameThatHasError === field.name) {
```

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+
+      expect(shouldBeDisabled).toBe(false);
+    });
+  }
+  for (const { questionType, label } of questionTypes) {
+    test(`should remain enabled when 'disableOnPrefill' is false for ${label} (${questionType})`, () => {
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:93]:
```suggestion
    test(`should return `false` when 'disableOnPrefill' is `false` for ${label} (${questionType})`, () => {
```

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+        value: "",
+        disableOnPrefill: false,
+        type: questionType,
+        name: `test-${questionType}-n`,
+        label: `test-${questionType}-l`,
+        placeholder: `test-${questionType}-p`,
+        required: true,
+      } as RhfFormField;
+
+      const formState = getFormatStateForQuestionType(questionType);
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:104]: Due to this formState having error, the code flow doesn't even check `disableOnPrefill` being false.

If someone were to introduce this bug where they return true from here, this test should have failed but it doesn't
image

File: packages/ui/components/formbuilder/formBuilderField.test.tsx [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+      //no errors
+      const formState = {};
+      const mockFormContext = { formState, control: {} as Control };
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+      expect(shouldBeDisabled).toBe(true);
+    });
+  }
+});
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/formBuilderField.test.tsx:133]: We need a test for `searchParams` as well. Input is only disabled if there are matching query params

File: packages/ui/components/formbuilder/testUtils.ts [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+    fireEvent.click(screen.getByTestId(`select-option-${props.questionType}`));
+    fireEvent.change(screen.getAllByRole("textbox")[0], { target: { value: props.identifier } });
+    fireEvent.change(screen.getAllByRole("textbox")[1], { target: { value: props.label } });
+    clickDisablePrefillBasedOnUserInput(props?.disableOnPrefill);
+    fireEvent.click(screen.getByTestId("field-add-save"));
+    await waitFor(() => {
+      expect(screen.queryByTestId(`field-${props.identifier}`)).toBeInTheDocument();
+    });
+  },
+  editQuestion: async (props: {
```
Comment (hariombalhara) [inline packages/ui/components/formbuilder/testUtils.ts:135]: I see similar code in here. We must reuse and add to the existing code if needed.

image

**Review by hariombalhara — CHANGES_REQUESTED** [review]
@rajeshj11 Thanks for the tests !! Left some more suggestions on them

**Comment by rajeshj11** [comment]
@anikdhabal @hariombalhara

**Comment by hariombalhara** [comment]
Here is the bookingForm 
image

**Comment by rajeshj11** [comment]
Thanks a lot.

**Comment by rajeshj11** [comment]
@anikdhabal @hariombalhara Can you please review the unit-test cases. moving the files and handling the duplicate code can be done later.(won't take more time)

File: packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts [hunk 25]
```diff
@@ -23,6 +27,107 @@ type UseInitialFormValuesProps = {
   };
 };
 
+//basically it will partially test the input values against the schema. validation are not needed at the initial stage
+export function getFieldSchema(field: Field): ZodSchema<unknown> {
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts:31]: Why is this `getFieldSchema` needed now?

File: packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts [hunk 26]
```diff
@@ -23,6 +27,107 @@ type UseInitialFormValuesProps = {
...
+    return { success: true, value: parsedResponses.data } as {
+      success: boolean;
+      value: string[] | string | undefined;
+    };
+  } else {
+    return { success: false, value: undefined } as {
+      success: boolean;
+      value: string[] | string | undefined;
+    };
+  }
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts:128]: I really don't understand why we have to reparse by a new schema. `parsedQuery` is supposed to be already parsed by the response schema

**Review by hariombalhara — DISMISSED** [review]
@rajeshj11 Left a few comments.

**Comment by rajeshj11** [comment]
@hariombalhara @anikdhabal 
I wanted to follow up on this pull request. Could you please review the latest comments at your earliest convenience? If this pull request is no longer needed, kindly close it.

File: packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts (hunk 25 above)
Comment (hariombalhara) [reply packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts:31]: @rajeshj11 Could you please check this. This new change is unnecessarily making things complex.

I don't understand the motivation for this change.

**Comment by hariombalhara** [comment]
I am taking over this PR as the priority has increased.

**Comment by rajeshj11** [comment]
> I am taking over this PR as the priority has increased.

if needed any assistance. please let me know

File: packages/prisma/schema.prisma [hunk 27]
```diff
@@ -1,4 +1,4 @@
-// This is your Prisma schema file,
+// This is your Prisma schema file
```
Comment (hariombalhara) [inline packages/prisma/schema.prisma:1]: For weird cache busting for failing tests,.

**Review by hariombalhara — DISMISSED** [review]

**Review by hariombalhara — APPROVED** [review]
